Configuration for the Python Extension's deployment via pip install

agurwicz opened the pull request.

agurwicz committed:
Configuration for the Python Extension's deployment via pip install

lendres reviewed (commented):
What changed in the setup file?  Or did this not exist before?

agurwicz commented:
> What changed in the setup file? Or did this not exist before?

The setup file already existed, just refactored it and moved part of the contents to `pyproject.toml` to keep with PEP convention.
Also moved it's location to the main directory (and changed paths accordingly in it) so that we can run `pip install` without the need to pass extra paths.

lendres commented:
Ok, please don't use this style/spacing for the imports:
``` Python
import os
from setuptools import setup, Extension
from pybind11 import get_include
```

Please keep the original style/spacing and keep every import on a separate line.  This is far more readable.
``` Python
import os
from   setuptools                     import setup
from   setuptools                     import Extension
from   pybind11                       import get_include
```

agurwicz committed:
Style convention fix
